src/battleship/Battleship.java

```diff
@@ -5,10 +5,26 @@
 final public class Battleship {
 
     final private Scanner scanner;
+    private List<Ship> playerOneShips;
+    private List<Ship> playerTwoShips;
+    private Board playerOneField;
+    private Board playerTwoField;
+    private Board playerOneHidden;
+    private Board playerTwoHidden;
+    private int playerOneShipsOnField;
+    private int playerTwoShipsOnField;
 
     public Battleship() {
 
         this.scanner = new Scanner(System.in);
+        this.playerOneShips = new ArrayList<>();
+        this.playerTwoShips = new ArrayList<>();
+        this.playerOneField = new Board();
+        this.playerTwoField = new Board();
+        this.playerOneHidden = new Board();
+        this.playerTwoHidden = new Board();
```

Comment:
These can all be field initialisers, there's no use of constructor arguments or logic in their initialisation.